### ISSUE TITLE
IndexError can also be raised by toml.loads() on invalid toml

### ISSUE THREAD

**Issue by pombredanne** [uiri/toml#156]
The docstring states:
```
        Raises:
            TypeError: When a non-string is passed
            TomlDecodeError: Error while decoding toml
```
This could be resolved either by:
re-raise a proper TomlDecodeError or updating the doctrsing

And example:
```
>>> import toml
>>> content = """[[source]
... verify_ssl = false
... """
>>> data = toml.loads(content)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/tmp/local/lib/python2.7/site-packages/toml.py", line 306, in loads
    if line[1].strip() != "":
IndexError: list index out of range

```

**Comment by pombredanne** [comment]
See also https://github.com/pyupio/dparse/pull/23 that triggered this issue

**Comment by uiri** [comment]
Fixed by 9e16f5b0

I will port this commit back to 0.9.X

**Comment by pombredanne** [comment]
@uiri Thank you.
